core/src/main/java/org/pegdown/ParserWithDirectives.java

```diff
@@ -162,7 +162,7 @@ public Node parseBlockContents(Var<Integer> markerLength, StringBuilderVar block
         // recursively parse inner contents
         Context<Object> context = getContext();
         Integer length = markerLength.get();
-        String contents = block.getString();
+        String contents = "\n" + block.getString() + "\n";
```

Comment:
Yes, it would have an impact but if you want to do something like tut you have to wrap internal content with "```" anyway to keep the preformatted text, for example the above would pass an HTML blockquote tag to the console directive.

BTW, I had missed that block directives already append a `\n` to the content so the fix can be simplified even further.

Anyway, if this is a showstopper I can submit a PR with tests and documentation instead.